src/strategies/token-bucket-strategy.ts

```diff
@@ -0,0 +1,156 @@
+import BaseStrategy from "./base-strategy";
+import { TokenBucketStrategyResponse, TokenBucketStrategyOptions } from "../interfaces";
+
+class TokenBucketStrategy extends BaseStrategy {
+  private bucketCapacity: number;
+  private refillRate: number; // Tokens per second
+
+  /**
+   * Constructor for TokenBucketStrategy
+   * @param options An instance of TokenBucketStrategyOptions
+   */
+  constructor(options: TokenBucketStrategyOptions) {
+    if (!options.store) {
+      throw new Error("A valid store implementation is required.");
+    }
+
+    super(options.store);
+
+    this.validateOptions(options);
+    this.bucketCapacity = options.bucketCapacity;
+    this.refillRate = options.refillRate;
+  }
+
+  /**
+   * Validates the options provided to the constructor.
+   * @param options The TokenBucketStrategyOptions to validate
+   */
+  private validateOptions(options: TokenBucketStrategyOptions): void {
+    if (options.bucketCapacity <= 0 || options.refillRate <= 0) {
+      throw new Error("bucketCapacity and refillRate must be greater than 0.");
```

Comment:
```suggestion
      throw new ConfigurationError("bucketCapacity and refillRate must be greater than 0.");
```
We can create custom errors and add the following reason in cause of the error